### ISSUE TITLE
🐛 [flutterfire] "Firebase.initializeApp" exception "Expected a value of type 'Object', but got one of type 'Null'"

### ISSUE THREAD

**Issue by mikexkllr** [FirebaseExtended/flutterfire#8128]
## Bug report

**An exception is thrown when initializing Firebase**
The error occurs on different platforms but occurs permanently when using Flutter Web. 

### Error Output

`  await Firebase.initializeApp(
    options: DefaultFirebaseOptions.currentPlatform,
  );`

This exception is thrown: 
`Expected a value of type 'Object', but got one of type 'Null'`

### Steps to reproduce

1. Start Flutter Chrome emulator (I use an arm64 Mac.)
2. Exception is thrown.

**Comment by darshankawar** [comment]
@mikexkllr 
Please provide details same as requested here: https://github.com/FirebaseExtended/flutterfire/issues/8132#issuecomment-1046800165

Also, which plugin are you trying to run that shows this error ? `firebase_auth` ?

**Comment by mikexkllr** [comment]
**This is the thrown exception on** 

`await Firebase.initializeApp( options: DefaultFirebaseOptions.currentPlatform, );
`
**Exception**

```
Exception has occurred.
"Error: Expected a value of type 'Object', but got one of type 'Null'
    at Object.throw_ [as throw] (http://localhost:55217/dart_sdk.js:5067:11)
    at Object.castError (http://localhost:55217/dart_sdk.js:5026:15)
    at Object.cast [as as] (http://localhost:55217/dart_sdk.js:5351:17)
    at Function.as (http://localhost:55217/dart_sdk.js:6576:31)
    at firebase_core_web.FirebaseCoreWeb.new.get [_ignoredServiceScripts] (http://localhost:55217/packages/firebase_core_web/firebase_core_web.dart.lib.js:142:65)
    at firebase_core_web.FirebaseCoreWeb.new._initializeCoreRequireJs (http://localhost:55217/packages/firebase_core_web/firebase_core_web.dart.lib.js:187:27)
    at _initializeCoreRequireJs.next (<anonymous>)
    at runBody (http://localhost:55217/dart_sdk.js:40590:34)
    at Object._async [as async] (http://localhost:55217/dart_sdk.js:40621:7)
    at firebase_core_web.FirebaseCoreWeb.new.[_initializeCoreRequireJs] (http://localhost:55217/packages/firebase_core_web/firebase_core_web.dart.lib.js:182:20)
    at firebase_core_web.FirebaseCoreWeb.new.initializeApp (http://localhost:55217/packages/firebase_core_web/firebase_core_web.dart.lib.js:214:47)
    at initializeApp.next (<anonymous>)
    at runBody (http://localhost:55217/dart_sdk.js:40590:34)
    at Object._async [as async] (http://localhost:55217/dart_sdk.js:40621:7)
    at firebase_core_web.FirebaseCoreWeb.new.initializeApp (http://localhost:55217/packages/firebase_core_web/firebase_core_web.dart.lib.js:210:20)
    at initializeApp (http://localhost:55217/packages/firebase_core/firebase_core.dart.lib.js:100:59)
    at initializeApp.next (<anonymous>)
    at runBody (http://localhost:55217/dart_sdk.js:40590:34)
    at Object._async [as async] (http://localhost:55217/dart_sdk.js:40621:7)
    at Function.initializeApp (http://localhost:55217/packages/firebase_core/firebase_core.dart.lib.js:99:20)
    at main (http://localhost:55217/packages/cnow_backbone/main.dart.lib.js:226:36)
    at main.next (<anonymous>)
    at runBody (http://localhost:55217/dart_sdk.js:40590:34)
    at Object._async [as async] (http://localhost:55217/dart_sdk.js:40621:7)
    at main$ (http://localhost:55217/packages/cnow_backbone/main.dart.lib.js:218:18)
    at main (http://localhost:55217/web_entrypoint.dart.lib.js:36:29)
    at main.next (<anonymous>)
    at http://localhost:55217/dart_sdk.js:40571:33
    at _RootZone.runUnary (http://localhost:55217/dart_sdk.js:40441:59)
    at _FutureListener.thenAwait.handleValue (http://localhost:55217/dart_sdk.js:35363:29)
    at handleValueCallback (http://localhost:55217/dart_sdk.js:35931:49)
    at Function._propagateToListeners (http://localhost:55217/dart_sdk.js:35969:17)
    at _Future.new.[_completeWithValue] (http://localhost:55217/dart_sdk.js:35817:23)
    at http://localhost:55217/dart_sdk.js:34998:46
    at _RootZone.runUnary (http://localhost:55217/dart_sdk.js:40441:59)
    at _FutureListener.then.handleValue (http://localhost:55217/dart_sdk.js:35363:29)
    at handleValueCallback (http://localhost:55217/dart_sdk.js:35931:49)
    at Function._propagateToListeners (http://localhost:55217/dart_sdk.js:35969:17)
    at _Future.new.[_completeWithValue] (http://localhost:55217/dart_sdk.js:35817:23)
    at async._AsyncCallbackEntry.new.callback (http://localhost:55217/dart_sdk.js:35838:35)
    at Object._microtaskLoop (http://localhost:55217/dart_sdk.js:40708:13)
    at _startMicrotaskLoop (http://localhost:55217/dart_sdk.js:40714:13)
    at http://localhost:55217/dart_sdk.js:36191:9"
```

**Comment by darshankawar** [comment]
@mikexkllr 
Thanks for the update. I tried the `firebase_auth` plugin example with flutterfire configured firebase app and ran it on web using `flutter run -d chrome`, which compiled and ran properly without any issues:

```
Running "flutter pub get" in example...                          2,690ms
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             18.5s
This app is linked to the debug service: ws://127.0.0.1:52236/u6r_llAgc9A=/ws
Debug service listening on ws://127.0.0.1:52236/u6r_llAgc9A=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:52236/u6r_llAgc9A=
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:52236/u6r_llAgc9A=
WARNING: You are using the Auth Emulator, which is intended for local testing only.  Do not use with production credentials.


```

```
[✓] Flutter (Channel stable, 2.10.2, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-GB)
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
[✓] Chrome - develop for the web
[✓] VS Code (version 1.62.0)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.
```

Can you try the same plugin [example](https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_auth/firebase_auth/example) and see if you still get the same error using it ?

**Comment by mikexkllr** [comment]
I tried the example project and it works. I guess I just used some newer dependencies. I gonna downgrade i guess it works well then.
